File: solutions/csharp/matrix/1/Matrix.cs
using System;

public class Matrix
{
    int[][] matrix;
    public Matrix(string input)
    {
        string[] rows = input.Split("\n");
        int numColumns = rows[0].Split(" ").Length;
        matrix = new int[rows.Length][];
        for (int i = 0; i < rows.Length; i++)
        {
            string[] columns = rows[i].Split(" ");
            matrix[i] = new int[numColumns];
            for (int j = 0; j < numColumns; j++)
            {
                matrix[i][j] = int.Parse(columns[j]);
            }
        }
    }

    public int[] Row(int row)
    {
        row--;
        return matrix[row];
    }

    public int[] Column(int col)
    {
        col--;
        int[] column = new int[matrix.Length];
        for (int i = 0; i < matrix.Length; i++)
        {
            column[i] = matrix[i][col];
        }
        return column;
    }
}

File: solutions/python/matrix/1/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.matrix_string = matrix_string
        self.rows = matrix_string.splitlines()
        self.list_rows = list(
            map(lambda x: list(map(int, x.split(' '))), self.rows))
        self.columns = [[] for i in range(len(self.list_rows[0]))]
        for row in self.list_rows:
            item_num = 0
            for item in row:
                self.columns[item_num].append(item)
                item_num += 1

    def row(self, index):
        return self.list_rows[index - 1]

    def column(self, index):
        return self.columns[index - 1]


Matrix("1 2\n3 4")
